File: Atividade_6/Java/ex02.java
/*
 * Entradas:
 * 1. Número de termos para calcular pi
 */

import java.util.Scanner;

public class ex02 {

    public static void main(String[] args) {
        System.out.println("Informe o número de termos para calcular pi: ");
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();

        double pi = Math.pow(calculaPi(n), 1.0/3.0);
        String piFormatted = String.format("%.5f", pi);
        System.out.println("O valor aproximado de pi com " + n + " termos é: " + piFormatted);
        
        in.close();
    }

    public static double calculaPi(int n) {
        double soma = 0.0;
        int denominador = 1;

        for (int i = 0; i < n; i++) {
            if (i % 2 == 0) {
                soma += 1.0 / (double) (denominador * denominador * denominador);
            } else {
                soma -= 1.0 / (double) (denominador * denominador * denominador);
            }
            denominador += 2;
        }
        return soma * 32;
    }
}

File: Atividade_6/Haskell/ex02.hs
-- Função para calcular pi com base no número de termos
calculaPi :: Int -> Double
calculaPi n = (somaPi 0 n 1 1) * 32

-- Função auxiliar para somar os termos para calcular pi
somaPi :: Double -> Int -> Int -> Int -> Double
somaPi soma 0 _ _ = soma
somaPi soma n denominador sinal
    | sinal == 1 = somaPi (soma + (1.0 / fromIntegral (denominador * denominador * denominador))) (n - 1) (denominador + 2) (-sinal)
    | otherwise  = somaPi (soma - (1.0 / fromIntegral (denominador * denominador * denominador))) (n - 1) (denominador + 2) (-sinal)

-- Função principal
main :: IO ()
main = do
    putStrLn "Informe o número de termos para calcular pi: "
    n <- readLn :: IO Int
    let piAproximado = (calculaPi n) ** (1.0/3.0)
    putStrLn $ "O valor aproximado de pi com " ++ show n ++ " termos é: " ++ show piAproximado
